spec/features/navigation_bar_spec.rb

```diff
@@ -1,17 +1,23 @@
 require 'rails_helper'
 feature "Navigation bar" do 
-  scenario "has link sign in and home link when user is not signed on" do 
+  before(:example) do |example|
+    unless example.metadata[:skip_before]
+      @user = create(:user)
+      manual_sign_in(@user)
+      visit root_path
+    end
+  end
+
+  scenario "has sign in link, sign up link, and home link when user is not signed in", :skip_before do 
     visit root_path
     expect(page.body).to have_link("Home")
     expect(page.body).to have_link("Sign in") 
+    expect(page.body).to have_link("Sign up")
   end
-  scenario "has account and sign out link when user is signed in " do 
-    visit root_path
-    user = create(:user)
-    manual_sign_in(user)
+  scenario "has topics, account, and sign out link when user is signed in " do 
```

Comment:
Take out some of the detail and summarize the scenario description. Leave the specifics to the code inside the test.